beacon_node/store/src/impls/beacon_state.rs

```diff
@@ -18,13 +18,17 @@ pub fn store_full_state<E: EthSpec>(
     Ok(())
 }
 
+// FIXME(tree-states): delete/move to migration
 pub fn get_full_state<KV: KeyValueStore<E>, E: EthSpec>(
     db: &KV,
     state_root: &Hash256,
     spec: &ChainSpec,
 ) -> Result<Option<BeaconState<E>>, Error> {
     let total_timer = metrics::start_timer(&metrics::BEACON_STATE_READ_TIMES);
 
+    // Here we need to figure out if the state at `state_root` is stored as a diff or snapshot and
+    // recurse
+
```

Comment:
Comment is out of date because we have this logic now in `load_hot_state`

```suggestion
```

Reply:
Fixed with a5825e1f9